modules/guardian/guardian.cc

```diff
@@ -60,14 +60,22 @@ void Guardian::Stop() { timer_.stop(); }
 
 void Guardian::OnTimer(const ros::TimerEvent&) {
   ADEBUG << "Timer is triggered: publish Guardian result";
-  std::lock_guard<std::mutex> lock(mutex_);
-  if (system_status_.has_safety_mode_trigger_time() && FLAGS_guardian_enabled) {
+  bool safety_mode_triggered = false;
+  {
+    std::lock_guard<std::mutex> lock(mutex_);
+    if (system_status_.has_safety_mode_trigger_time()) {
+      safety_mode_triggered = true;
+    }
+  }
+
+  if (safety_mode_triggered && FLAGS_guardian_enabled) {
```

Comment:
move FLAGS_guardian_enabled to the front of the function to do quick rejection, or is this gflag really needed?

In addition, if all the ROS calls use the same mutex_, it is equivalent to using single thread receiving mode in ROS